DEVGUIDE.md

```diff
@@ -0,0 +1,30 @@
+# Developer's Guide to Service-Catalog
+
+## Building
+
+To build the service-catalog you have two options:
+* `make build`
+* `DOCKER=1 make build`
+
+Both will build all of the executables, into the `bin` directory. However,
+the second option will do the build within a Docker container - meaning you
+do not need to have all of the necessary tooling installed on your host
+(such as a golang compiler or glide). Whichever option you choose, the
+results should be the same.
```

Comment:
I didn't touch on the multi-arch build stuff yet - wanted to wait until we're sure we want to head this way.
How does it work today? What are the default exe? the local arch/os or linux?